### PR TITLE
add graph data format design

### PR DESCRIPTION
**Pull request by jacquesqiao**

### CODE REVIEW

File: docs/Graph Data Format.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+## VisualDL Graph data format design
```
Comment (Superjomn) [inline docs/Graph Data Format.md]: Data Format Design

File: docs/Graph Data Format.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+## Backgroud
+Neural network have a concept of `Model`. VisualDL has a very import feature that it can visualize the structure of deep learning `Model`. As an visualize tool that want to support many different deeplearning framework, we should use a generic Model data format that is supported by most of frameworks.
```
Comment (Superjomn) [inline docs/Graph Data Format.md]: import -> important

Grammarly

Comment (Superjomn) [reply docs/Graph Data Format.md]: Model -> model

visualize tool -> vasualization tool

File: docs/Graph Data Format.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+Facebook opensourced a [ONNX](http://onnx.ai/)(Open Neural Network Exchange). ONNX is a open format to represent deep learning models. It now officially supported by world's most popular framework like Tensorflow, Caffe2, MxNet, PaddlePaddle Fluid, so we decided to choose ONNX as the standard Model format and support the visualization of ONNX model.
```
Comment (Superjomn) [inline docs/Graph Data Format.md]: a -> an?

MxNet -> MXNet

> It now officially supported by world's most popular framework

Currently, it is officially supported by most of the popular frameworks in the world.

File: docs/Graph Data Format.md (hunk 1 above)
Comment (jacquesqiao) [reply docs/Graph Data Format.md]: done

File: docs/Graph Data Format.md (hunk 2 above)
Comment (jacquesqiao) [reply docs/Graph Data Format.md]: done

File: docs/graph_data_format.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+## Rest API data format
+Frontend use rest API to get data from the server, the data format will be JSON, the data structure of a Graph is as below. Each Graph will have three vectors inside:
+
+- `node` represent Operator. It has type, input, and output.
+- `input` represent input data or parameters, it has shape info.
+- `output` represent output data of the graph, it has shape info.
+
+```json
+{
+    "graph": {
```
Comment (deqingli) [inline docs/graph_data_format.md:22]: 最外面这层可以不要

Comment (deqingli) [reply docs/graph_data_format.md:22]: 我说的是 "graph" : {} 这个。可以直接 {"nodes": [], "input": [],...}  这样

**Review by deqingli — COMMENTED** [review]
整体是可以的，做外层的 graph 可以不要，详见评论

File: docs/graph_data_format.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                "output": [
+                    "out3"
+                ]
+            }
+        ],
+        "inputs": [
+            {
+                "name": "in1",
+                "shape": [
+                    -1,
```
Comment (daming-lu) [inline docs/graph_data_format.md:52]: shape是矩阵的维度，应该不能为负吧？

File: docs/graph_data_format.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+                    2,
+                    3
+                ]
+            }
+        ],
+        "outputs": [
+            {
+                "name": "out3",
+                "shape": [
+                    -1,
```
Comment (daming-lu) [inline docs/graph_data_format.md:75]: 同上

**Review by daming-lu — CHANGES_REQUESTED** [review]

File: docs/graph_data_format.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
+## VisualDL Graph Data Format Design
+
+## Background
+Neural network has a concept of `Model`. VisualDL has a very important feature that it can visualize the structure of deep learning `Model`. As a visualization tool that wants to support many different deep learning frameworks, we should use a generic Model data format that is supported by most of the frameworks.
+
+Facebook open-sourced an [ONNX](http://onnx.ai/)(Open Neural Network Exchange). ONNX is an open format to represent deep learning models. It now officially supported by world's most popular framework like Tensorflow, Caffe2, MxNet, PaddlePaddle Fluid, so we decided to choose ONNX as the standard Model format and support the visualization of ONNX model.
```
Comment (daming-lu) [inline docs/graph_data_format.md:6]: open-sourced a project called ONNX?

Comment (daming-lu) [reply docs/graph_data_format.md:6]: It is now officially supported by world's most popular frameworks such as Tensorflow, Caffe2, MxNet, PaddlePaddle Fluid, etc. So we decided to choose ONNX as the standard Model format and support the visualization of ONNX model.

File: docs/graph_data_format.md [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+
+## IR of ONNX
+The description of ONNX IR can be found [here](https://github.com/onnx/onnx/blob/master/docs/IR.md). The most important part is the definition of [Graph](https://github.com/onnx/onnx/blob/master/docs/IR.md#graphs).
+
+Each computation data flow graph is structured as a list of nodes that form a graph. Each node is a call to an operator. Nodes have zero or more inputs, one or more outputs, and zero or more attribute-value pairs.
```
Comment (daming-lu) [inline docs/graph_data_format.md:11]: Each node is called an operator.

File: docs/graph_data_format.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+Facebook open-sourced an [ONNX](http://onnx.ai/)(Open Neural Network Exchange). ONNX is an open format to represent deep learning models. It now officially supported by world's most popular framework like Tensorflow, Caffe2, MxNet, PaddlePaddle Fluid, so we decided to choose ONNX as the standard Model format and support the visualization of ONNX model.
+
+## IR of ONNX
+The description of ONNX IR can be found [here](https://github.com/onnx/onnx/blob/master/docs/IR.md). The most important part is the definition of [Graph](https://github.com/onnx/onnx/blob/master/docs/IR.md#graphs).
+
+Each computation data flow graph is structured as a list of nodes that form a graph. Each node is a call to an operator. Nodes have zero or more inputs, one or more outputs, and zero or more attribute-value pairs. 
+
+## Rest API data format
+Frontend use rest API to get data from the server, the data format will be JSON, the data structure of a Graph is as below. Each Graph will have three vectors inside:
```
Comment (daming-lu) [inline docs/graph_data_format.md:14]: Frontend uses RESTful API to get data from the server. The data format will be JSON. The data structure of a Graph is as below.

File: docs/graph_data_format.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+- `node` represent Operator. It has type, input, and output.
```
Comment (daming-lu) [inline docs/graph_data_format.md:16]: represents.  我们是否要说明 shape 就是矩阵的维度？

File: docs/graph_data_format.md (hunk 5 above)
Comment (jacquesqiao) [reply docs/graph_data_format.md:52]: 负值表示这个维度不确定，这个是数据的batch_size

File: docs/graph_data_format.md (hunk 4 above)
Comment (jacquesqiao) [reply docs/graph_data_format.md:22]: 好的

File: docs/graph_data_format.md (hunk 10 above)
Comment (jacquesqiao) [reply docs/graph_data_format.md:16]: 一般来讲，shape就是矩阵的维度

**Review by daming-lu — APPROVED** [review]

File: docs/graph_data_format.md (hunk 5 above)
Comment (daming-lu) [reply docs/graph_data_format.md:52]: 明白了，这里是读入数据的行数，可多可少，是个变量。谢谢乔老师。

**Pull request merged**